core/src/main/scala/Github.scala

```diff
@@ -83,35 +85,67 @@ object Github {
     zen: String
   ) extends Event
 
+  /**
+   * Reference https://developer.github.com/v3/activity/events/types/#pullrequestevent
+   *
+   * This event definition is a placeholder for future support for inbound pull request
+   * deployments. This skeleton is being provided so that Nelson can parse the inbound
+   * payload and give a 200 OK response so Github doesn't think Nelson is faulty.
+   */
+  final case class PullRequestEvent(
+    id: Long,
+    url: String,
+    slug: Slug
+  ) extends Event
+
+  /**
+   * Reference: https://developer.github.com/v3/activity/events/types/#releaseevent
+   *
+   * Nelson still accepts these as inbound input for historical reasons, so we
+   * simply accept it on the webhook entrypoint so that Github does not get invalid
+   * responses from Nelson (putting the webhook into an unsafe state).
+   */
   final case class ReleaseEvent(
     id: Long,
     slug: Slug,
     repositoryId: Long
   ) extends Event
 
+  /**
+   * Reference: https://developer.github.com/v3/activity/events/types/#deploymentevent
+   */
+  final case class Deployment(
+    id: Long,
+    slug: Slug,
+    repositoryId: Long,
+    ref: String, // sha, branch or tag
+    environment: String,
+    deployables: List[Manifest.Deployable],
+    url: String
+  ) extends Event {
+    def findDeployable(withName: String): IO[Manifest.Deployable] =
+      deployables.find(_.name.trim.toLowerCase == withName.trim.toLowerCase)
+        .fold[IO[Manifest.Deployable]](IO.raiseError(new ProblematicDeployable(s"could not find an asset named $withName", url)))(IO.pure)
+  }
+
+  // NOTE(timperrett): this is legacy, but we're keeping it around so that
+  // our webhooks still give valid responses to Github in the event that
+  // operators of Nelson forget to disable the release webhook.
   final case class Release(
```

Comment:
Mark as `@deprecated("Release triggers are deprecated in favor of Deployments.", "0.12.0")`

Reply:
Actually we cannot do this, as we compile lifting warnings into errors.